libbeat/scripts/generate_template.py

```diff
@@ -292,12 +293,12 @@ def fill_field_properties(args, field, defaults, path):
             path = field["name"]
         prop, dynamic = fill_section_properties(args, field, defaults, path)
 
+        properties[field.get("name")] = {
+            "type": "nested",
```

Comment:
I am not sure about this one. My understanding of the elasticsearch documentation is, without the definition of the `type` it would become type [object](https://www.elastic.co/guide/en/elasticsearch/reference/5.1/object.html).
Because the user explicitly defined the field type to be `nested`, setting the type implicitly to object would be wrong and therefore we are required to explicitly define the type as [nested](https://www.elastic.co/guide/en/elasticsearch/reference/5.1/nested.html).
Either way I think we are on the safe side, if we keep the `type: nested`, because it was there and therefore we do not break backwards compatibility.